src/ansi.rs

```diff
@@ -770,10 +778,104 @@ pub enum CursorShape {
     Hidden,
 }
 
-/// Terminal modes.
+/// Wrapper for the public mode.
+#[derive(Debug, Clone, Copy, Eq, PartialEq)]
+pub enum PublicMode {
+    /// Known public mode.
+    Named(NamedPublicMode),
+    /// Unidentified publc mode.
+    Unidentified(u16),
```

Comment:
I think `Unknown` is the more common name for things like this and it's also shorter.